### ISSUE TITLE
director_scheduler fails during director DB migration

### ISSUE THREAD

**Issue by beyhan** [cloudfoundry/bosh#2155]
**Describe the bug**
`director_scheduler` will fail until the director DB migration is finished. Afterwards the process seem to work properly. 

**To Reproduce**
1. Add a sleep in director [start script](https://github.com/cloudfoundry/bosh/blob/641ea9493cd4e32361d8496457f8cb33afc2175e/jobs/director/templates/director#L7-L8) before DB migration.
2. Create a dev release with this change
2. Deploy a bosh director on any IaaS
3. Check the stdout and stderr logs of `director_scheduler`

**Expected behavior**
`director_scheduler` process does not fail with unexpected exception

**Logs**
```
/var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:152:in `async_exec': PG::UndefinedTable: ERROR:  relation "director_attributes" does not exist (Sequel::DatabaseError)
LINE 1: SELECT * FROM "director_attributes" WHERE ("name" = 'uuid') ...
                      ^
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:152:in `block in execute_query'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/database/logging.rb:43:in `log_connection_yield'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:152:in `execute_query'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:140:in `block in execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:116:in `check_disconnect_errors'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:140:in `execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:496:in `_execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:314:in `block (2 levels) in execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:518:in `check_database_errors'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:314:in `block in execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/database/connecting.rb:253:in `block in synchronize'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/connection_pool/threaded.rb:92:in `hold'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/database/connecting.rb:253:in `synchronize'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:314:in `execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/dataset/actions.rb:1083:in `execute'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/adapters/postgres.rb:610:in `fetch_rows'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/dataset/actions.rb:935:in `with_sql_each'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/dataset/actions.rb:945:in `with_sql_first'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/dataset/actions.rb:701:in `single_record!'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/dataset/actions.rb:244:in `first'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/sequel-5.16.0/lib/sequel/model/plugins.rb:33:in `first'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/models/director_attribute.rb:8:in `find_or_create_uuid'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/config.rb:195:in `configure'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/config.rb:560:in `configure_evil_config_singleton!'
        from /var/vcap/data/packages/director/84c2f2a4e72f95ae4fde5cb64a4dd04bb75ffec4/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/lib/bosh/director/app.rb:25:in `initialize'
        from /var/vcap/packages/director/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/bin/bosh-director-scheduler:17:in `new'
        from /var/vcap/packages/director/gem_home/ruby/2.4.0/gems/bosh-director-0.0.0/bin/bosh-director-scheduler:17:in `<top (required)>'
        from /var/vcap/packages/director/bin/bosh-director-scheduler:29:in `load'
        from /var/vcap/packages/director/bin/bosh-director-scheduler:29:in `<main>'
```

**Versions (please complete the following information):**
 - BOSH version 268.4+ (most probably also with earlier versions, but didn't test it)

**Additional context**
`director_scheduler` should wait before accessing the Sequel models, which are based on director DB until the director DB is initialized properly.  E.g. the same has been done for `director_sync_dns` [in this pr](https://github.com/cloudfoundry/bosh/pull/2154).

@langered

**Comment by dpb587-pivotal** [comment]
Doesn't this fail before `monit` considers it healthy, and doesn't `monit` automatically restart it when this happens? I'd expect the impact of this to be minimal. Is the goal to minimize log noise, or is there some functional impact from it?

**Comment by beyhan** [comment]
This is more for documenting reasons and to have it in mind if we change something in the `scheduler`. There is no functional impact at the moment because as you mentioned `monit` restarts the process. E.g. we discovered a similar situation in `bosh-director-sync-dns`, where a combination with EventMachine and Thread spawning shadows the exception and the process doesn't fails. There we proposed this [fix](https://github.com/cloudfoundry/bosh/pull/2154/commits/707534cac2b85f22929ce8250ce29a8024c3dc35)
